### PR TITLE
BrowserStack migration

### PR DESCRIPTION
**Pull request by carhartl**
Moving from Saucelabs to BrowserStack ~, so we can automate testing in prehistoric Internet Explorers (6 + 7) again~. BrowserStack kindly provides access for open-source projects ❤️

Overall using BrowserStack seems a little more streamlined (removing code!).

- [x] Travis/Grunt setup
- [x] Finalize list of browsers to test in (=> `browserstack.json`)
- [x] Adapt readme (badge)
- [x] Add shoutout

### CODE REVIEW

File: travis.sh [hunk 1]
```diff
@@ -1,5 +1,8 @@
 #!/bin/bash
-if [ -z "$SAUCE_ACCESS_KEY" ]; then
+
+# Only use `grunt ci` for commits pushed to this repo. Fall back to regular test
+# for pull requests (as secure variables won't be exposed there).
```
Comment (carhartl) [inline travis.sh]: Seems no longer correct...

**Comment by carhartl** [comment]
Tests in Travis are already running...

**Comment by carhartl** [comment]
The browserstack runner in Travis says `All tests done, failures: 3.`, even though I looked and all tests pass... ?

**Comment by carhartl** [comment]
Ah, no tests ran for `"ie_6", "ie_7", "ie_8"` (making it missing 3), might be that we can't use this short config notation for those...

**Comment by carhartl** [comment]
In these timed out IE test runs there was another error it seems, again in `afterEach`: https://www.browserstack.com/automate/builds/3178819721f9cfe05d6854c454f248b30ff7b2af/sessions/dbc4fe311f356f2b2cc2254519acb4266e52c81d

**Comment by carhartl** [comment]
Interesting.. BrowserStack seems quite volatile. Now I'm getting `Error: "expires option as days from now" failed, "c=v; path=/; expires=Sun, 11 Mar 2018 09:27:47 GMT" includes "expires=Sun, 11 Mar 2018 08:27:47 GMT"` (one hour difference!) all over the place, but didn't see that yesterday.

**Comment by carhartl** [comment]
Another reason to resume working on this... I found out that it's no longer possible to run tests in mobile browsers using Saucelabs (at least with grunt-saucelabs, which accesses their rest api) => 3138f588a1a7f3f6ec4012185104bcb974702441

**Comment by FagnerMartinsBrack** [comment]
Damn...

**Comment by carhartl** [comment]
Resumed work on the migration.. in IE 6 getting the same error described here: https://github.com/browserstack/browserstack-runner/issues/161 

(posting for reference)

**Comment by carhartl** [comment]
bildschirmfoto 2018-04-13 um 20 59 44

**Comment by carhartl** [comment]
So.. that's that. The problem seems to be within browserstack-runner (seeing the same error even with an almost blank page containing no tests at all nor QUnit).

**Comment by carhartl** [comment]
Option 1: Fork browserstack-runner + JSON3 to fix running BrowserStack Automate w/ IE 6 - 8.

Option 2: Skip running testing these browsers in Automate/CI, but at least have a way to run tests in mobile browsers, plus an easy way (BrowserStack Live) to test these IEs manually.

Option 3: Something I've overlooked.

**Comment by FagnerMartinsBrack** [comment]
Option 1 seems reasonable.

**Comment by FagnerMartinsBrack** [comment]
@carhartl I'm in the process of triaging, can I close this PR?

File: test/tests.js [hunk 2]
```diff
@@ -117,7 +117,7 @@ QUnit.test('Call to read cookie when there is another unrelated cookie with malf
 	document.cookie = 'c=v';
 	assert.strictEqual(Cookies.get('c'), 'v', 'should not throw a URI malformed exception when retrieving a single cookie');
 	assert.deepEqual(Cookies.get(), { c: 'v' }, 'should not throw a URI malformed exception when retrieving all cookies');
-	Cookies.withConverter(unescape).remove('invalid');
+	document.cookie = 'invalid=foo; expires=Thu, 01 Jan 1970 00:00:00 GMT';
```
Comment (FagnerMartinsBrack) [inline test/tests.js]: Why this change?

Comment (FagnerMartinsBrack) [reply test/tests.js]: Oh, probably due to [this comment](https://github.com/js-cookie/js-cookie/pull/409#issuecomment-366504267). It's been so long I lost context of this PR sorry :D

**Comment by carhartl** [comment]
Looks related: https://github.com/browserstack/browserstack-runner/commit/826d2c7c9edbc442c1b3b0256f4115d7962bda15#diff-bc2ed23c09803d81e8a2abd1fc9d30ee

**Comment by carhartl** [comment]
https://github.com/browserstack/browserstack-runner/issues/212

**Comment by carhartl** [comment]
I think I'm not going to sink any more time into this though... going to transfer anything useful I found out along the way to master. In particular I believe https://github.com/js-cookie/js-cookie/pull/409/commits/d9bfab0233312b407580989d60f1b90e304d453c will be desired, since at the moment, due to the more recently introduced node versions build matrix (https://github.com/js-cookie/js-cookie/commit/5e969423792fb2adaa1eaeca811ce3129866327e#diff-354f30a63fb0907d4ad57269548329e3) we're running tests 3 times in saucelab, where one is sufficient... @FagnerMartinsBrack

**Comment by carhartl** [comment]
PS: Apparently, nobody in this world is running automated tests in IE 6 - 9 any longer...

**Comment by carhartl** [comment]
I'm starting to wonder though, why are we even running tests in several node versions instead of just say the latest LTS, given we're (exclusively?) targeting client-side/browser environments... is this to ensure build scripts compatilibity?

**Comment by carhartl** [comment]
Aha, using https://github.com/paulmillr/console-polyfill I can workaround this problem...

**Comment by carhartl** [comment]
We‘re still facing the JSON3 problem in IE 6-8, forking browserstack-runner + JSON3 does no longer look like a feasible thing to do to me. We‘d be using our time more efficiently by releasing a mayor version and officially drop support for IE 6-8 (at least).

**Comment by FagnerMartinsBrack** [comment]
@carhartl I've added the comment: https://github.com/js-cookie/js-cookie/issues/370#issuecomment-451100724. Any chances we can have tests for IE 9+ in BrowserStack? Any significant change to js-cookie means we need to drop oldIE and therefore a major bump?

**Comment by FagnerMartinsBrack** [comment]
Of course, assuming everything else works fine with BrowserStack and the only problem is with oldIE (IE 6 to 8)

**Comment by carhartl** [comment]
> Any chances we can have tests for IE 9+ in BrowserStack

We have now, adding the console polyfill fixed the problem in IE9. We can give it a try and reactivate all the other browsers to test in. And then, with Browserstack at least we do still have the ability to test oldIE manually, something we can‘t do with Saucelabs.

**Comment by carhartl** [comment]
I.e. running tests manually in oldIE using Browserstack‘s “Live” offering works fine, only the browserstack-runner for using “Automate” injects scripts into the test page which are causing these problems. Just in case that wasn’t entirely clear.

**Comment by FagnerMartinsBrack** [comment]
I'm good to have Browserstack with IE 9+, what do you think @carhartl?

**Comment by carhartl** [comment]
> I'm good to have Browserstack with IE 9+, what do you think @carhartl?

Me too!

**Comment by carhartl** [comment]
I've added a couple of browsers back in and build is fine: https://travis-ci.org/js-cookie/js-cookie/builds/474792190

Probably not the final list:
```
    "browsers": [
        "chrome_latest",
        "chrome_previous",
        "firefox_latest",
        "firefox_previous",
        "ie_11",
        "ie_10",
        "ie_9",
        "opera_latest",
        {
            "browser": "safari",
            "browser_version": "latest",
            "os": "OS X",
            "os_version": "Mojave"
        },
        {
            "browser": "safari",
            "browser_version": "latest",
            "os": "OS X",
            "os_version": "High Sierra"
        }
    ]
```

**Comment by carhartl** [comment]
Testing on a real Android device works. Though I can configure an iPhone to test in, the runner doesn't get past launching. Tried with a couple of different iPhones and iOS versions...

screenshot 2019-02-19 at 11 21 03

**Comment by carhartl** [comment]
Closes this, I still can't seem to be able to run tests in an iOS device. At least not using https://github.com/browserstack/browserstack-runner
